pkg/apis/core/validation/project_test.go

```diff
@@ -163,30 +163,117 @@ var _ = Describe("Project Validation Tests", func() {
+					"Field":    Equal("spec.namespace"),
+					"BadValue": Equal("garden-foo%bar"),
+					"Detail":   Equal(`a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')`),
+				}))),
+			),
 			Entry("should allow Project with namespace garden",
 				"garden",
 				BeEmpty(),
 			),
 		)
 
-		It("should forbid Project specification with empty or invalid key for description", func() {
-			project.Spec.Description = ptr.To("")
+		DescribeTable("Project spec.description",
+			func(description *string, matcher gomegatypes.GomegaMatcher) {
+				project.Spec.Description = description
 
-			errorList := ValidateProject(project)
+				errorList := ValidateProject(project)
 
-			Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
-				"Type":  Equal(field.ErrorTypeRequired),
+				Expect(errorList).To(matcher)
+			},
+			Entry("should allow no description",
+				nil,
+				BeEmpty(),
+			),
+			Entry("should allow valid description",
+				ptr.To(`This is a project description, 庭師.`),
+				BeEmpty(),
+			),
+			Entry("should forbid Project with empty description",
+				ptr.To(""),
+				ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeRequired),
+					"Field": Equal("spec.description"),
+				}))),
+			),
+			Entry("should forbid Project description with special characters",
+				ptr.To("<foo>"),
+				ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeForbidden),
+					"Field": Equal("spec.description"),
+				}))),
+			),
+		)
+
+		It("should allow already accepted invalid project description", func() {
+			project.Spec.Description = ptr.To("<foo>")
+			newProject := prepareProjectForUpdate(project)
+
+			Expect(ValidateProjectUpdate(newProject, project)).To(BeEmpty())
+		})
+
+		It("should forbid Project description with special characters when updated", func() {
+			project.Spec.Description = ptr.To("foo")
+			newProject := prepareProjectForUpdate(project)
+			newProject.Spec.Description = ptr.To("<foo>")
+
+			Expect(ValidateProjectUpdate(newProject, project)).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+				"Type":  Equal(field.ErrorTypeForbidden),
 				"Field": Equal("spec.description"),
 			}))))
 		})
 
-		It("should forbid Project specification with empty or invalid key for purpose", func() {
-			project.Spec.Purpose = ptr.To("")
+		DescribeTable("Project spec.purpose",
+			func(purpose *string, matcher gomegatypes.GomegaMatcher) {
+				project.Spec.Purpose = purpose
 
-			errorList := ValidateProject(project)
+				errorList := ValidateProject(project)
 
-			Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
-				"Type":  Equal(field.ErrorTypeRequired),
+				Expect(errorList).To(matcher)
+			},
+			Entry("should allow no purpose",
```

Comment:
_nit:_ Same as https://github.com/gardener/gardener/pull/12902#discussion_r2329516815
```suggestion
			Entry("should allow unset purpose",
```